.github/workflows/tests.yml

```diff
@@ -396,3 +325,231 @@ jobs:
+      - run: echo "::add-matcher::.github/matchers/rust.json"
+      - run: make slim-builds
+      - run: make build-x86_64-unknown-linux-gnu
+
+  # TODO: Upgrade to test once the tz issues are solved.
+  # https://github.com/timberio/vector/issues/3771
+  build-aarch64-unknown-linux-gnu:
+    name: Build - aarch64-unknown-linux-gnu
+    runs-on: ubuntu-20.04
+    if: |
+      !contains(github.event.pull_request.labels.*.name, 'ci-condition: skip')
+    steps:
+      - uses: actions/checkout@v2
+      - run: make ci-sweep
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - run: sudo bash scripts/environment/bootstrap-ubuntu-20.04.sh
+      - run: bash scripts/environment/prepare.sh
+      - run: echo "::add-matcher::.github/matchers/rust.json"
+      - run: make slim-builds
+      - run: make build-aarch64-unknown-linux-gnu
+
+  build-deb-package:
+    name: Build .deb package
+    runs-on: ubuntu-latest
+    if: |
+      !contains(github.event.pull_request.labels.*.name, 'ci-condition: skip')
+    steps:
+      - uses: actions/checkout@v2
+      - run: make ci-sweep
+      - run: make package-deb-x86_64
+      - uses: actions/upload-artifact@v2
+        with:
+          name: e2e-test-deb-package
+          path: target/artifacts/*
+
+  # Github Actions don't support `matrix` at the job-level `if:` condition.
+  # We apply this workaround - compute `matrix` in a preceding job, and assign
+  # it's value dynamically at the actual test job.
+  # This approach can be advanced further by, for instance, dynamically
+  # detecting versions of various components, or reading them from `.meta`.
+  # See https://github.community/t/feature-request-and-use-case-example-to-allow-matrix-in-if-s/126067
+  compute-k8s-test-plan:
+    name: Compute K8s test plan
+    runs-on: ubuntu-latest
+    if: |
+      !contains(github.event.pull_request.labels.*.name, 'ci-condition: skip')
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/github-script@v2
+        id: set-matrix
+        with:
+          script: |
+            // Parameters.
+            const minikube_version = [
+              "v1.11.0", // https://github.com/kubernetes/minikube/issues/8799
+            ]
+            const kubernetes_version = [
+              { version: "v1.19.2", is_essential: true },
+              { version: "v1.18.9" },
+              { version: "v1.17.12" },
+              { version: "v1.16.15" }, // v1.16.13 is broken, see https://github.com/kubernetes/kubernetes/issues/93194
+              { version: "v1.15.12" },
+              { version: "v1.14.10" },
+            ]
+            const container_runtime = [
+              "docker",
+              "containerd",
+              "crio",
+            ]
+            const ci_condition_label = 'ci-condition: k8s e2e all targets'
```

Comment:
Will we be able to set the `ci-condition: k8s e2e all targets` with bors? Also, we might not need this anymore, and just run all the target all the time.

Reply:
I think we can pull it out. Feel free to remove the logic in `master`.